planner/core/testdata/partition_pruner_out.json

```diff
@@ -175,6 +175,41 @@
           "└─Selection 0.02 cop[tikv]  or(and(eq(test_partition.t7.a, 1), eq(test_partition.t7.b, 2)), and(eq(test_partition.t7.a, 3), eq(test_partition.t7.b, 4)))",
           "  └─TableFullScan 10000.00 cop[tikv] table:t7 keep order:false, stats:pseudo"
         ]
+      },
+      {
+        "SQL": "explain format = 'brief' select * from t8 where a between 10 and 11",
+        "Result": [
+          "TableReader 250.00 root partition:p4,p5 data:Selection",
+          "└─Selection 250.00 cop[tikv]  ge(test_partition.t8.a, 10), le(test_partition.t8.a, 11)",
+          "  └─TableFullScan 10000.00 cop[tikv] table:t8 keep order:false, stats:pseudo"
+
+        ]
+      },
+      {
+        "SQL": "explain format = 'brief' select * from t8 where (a is null) or (a between 5 and 8)",
+        "Result": [
+          "TableReader 260.00 root partition:p0,p1,p2,p5 data:Selection",
+          "└─Selection 260.00 cop[tikv]  or(isnull(test_partition.t8.a), and(ge(test_partition.t8.a, 5), le(test_partition.t8.a, 8)))",
+          "  └─TableFullScan 10000.00 cop[tikv] table:t8 keep order:false, stats:pseudo"
+
+        ]
+      },
+      {
+        "SQL": "explain format = 'brief' select * from t8 where a between 5 and 12",
+        "Result": [
+          "TableReader 250.00 root partition:all data:Selection",
+          "└─Selection 250.00 cop[tikv]  ge(test_partition.t8.a, 5), le(test_partition.t8.a, 12)",
+          "  └─TableFullScan 10000.00 cop[tikv] table:t8 keep order:false, stats:pseudo"
+
+        ]
+      },
+      {
+        "SQL": "explain format = 'brief' select * from t8 where (a <= 10 and a >= 8) or (a <= 13 and a >= 11) or (a <= 16 and a >= 14)",
+        "Result": [
+          "TableReader 750.00 root partition:p0,p1,p2,p3,p4,p5 data:Selection",
```

Comment:
Can we try to use `partition:all` instead of `partition:p0,p1,p2,p3,p4,p5`?

Reply:
This is try to not conflict with the existing testing case: [cbo_test.go:L1118](https://github.com/pingcap/tidb/blob/master/planner/core/cbo_test.go#L1118).

If we try to use `partition:all` if we determine all the partitions are used, this kind of existing testing cases will be changed too.

Reply:
Oh, that's so bad. How about creating an issue to record this problem so that we can fix it later.

Reply:
Sure, I'll do it later.